### ISSUE TITLE
Mel spectrogram format mismatch between training and HiFi-GAN vocoder causes noise output

### ISSUE THREAD

**Issue by stilletto** [igorshmukler/kokoro-ruslan#85]
## Problem

The mel spectrogram computation in `dataset.py:667` uses:

```python
mel_spec = torch.log(mel_spec_linear + 1e-9)
```

This produces mel values in range approximately `[-20, +8]`.

However, the HiFi-GAN universal vocoder (`generator_universal_v1`) was trained with standard spectral normalization:

```python
torch.log(torch.clamp(x, min=1e-5))  # range: [-11.5, ~0]
```

This mismatch means the vocoder receives mel values far outside its training distribution, producing noise/artifacts instead of speech — even when fed ground truth mel spectrograms from the dataset.

Additionally, `generator.py:115` and `inference.py:572` hard-clip mel output to `max=2.0`:

```python
mel_output.clamp_(min=-11.5, max=2.0)
```

The actual mel range from the dataset extends to +7.6 (with `1e-9` epsilon) or +0.9 (with `1e-5` clamp). Either way, `max=2.0` destroys high-energy speech frames when using the `1e-9` epsilon format.

## Impact

- Generated audio sounds like noise regardless of training progress
- Validation loss decreases normally (teacher-forced mel prediction works), but inference output is unintelligible
- GPT-4o-audio evaluation confirmed: 0/10 on all generated samples, comment: "only noise, no speech"

## Solution

1. **`dataset.py:667`** — change epsilon to match HiFi-GAN format:

```python
mel_spec = torch.log(torch.clamp(mel_spec_linear, min=1e-5))
```

2. **`generator.py:115`** and **`inference.py:572`** — widen or remove the clamp to match the new range:

```python
mel_output.clamp_(min=-12.0, max=2.0)  # matches log(clamp(x, min=1e-5)) range
```

3. **`dataset.py:654`** — the peak normalization (`audio = audio / max(abs(audio))`) shifts mel values by a per-sample constant, making the mel distribution inconsistent across samples with different peak amplitudes. Consider removing it for consistency with the vocoder.

## Mel parameters

- sample_rate=22050, n_fft=1024, hop_length=256, win_length=1024, n_mels=80, fmin=0, fmax=8000

## Environment

- PyTorch 2.8.0+cu128, torchaudio 2.8.0+cu128
- HiFi-GAN universal_v1 (pretrained, downloaded by VocoderManager)
- RUSLAN corpus, 13,965 samples
